### PR TITLE
[KEYCLOAK-12980] Username not updated when "Email as username" is ena…

### PR DESCRIPTION
**Pull request by douglaspalmer**
…bled

### CODE REVIEW

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/UserTest.java [hunk 1]
```diff
@@ -458,6 +458,18 @@ public void createUserWithoutUsername() {
...
     }
 
+    @Test
+    public void createUserWithEmailAsUsername() {
+        switchRegistrationEmailAsUsername(true);
+
+        String id = createUser();
+        UserResource user = realm.users().get(id);
+        UserRepresentation userRep = user.toRepresentation();
+        assertEquals("user1@localhost", userRep.getEmail());
```
Comment (stianst) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/UserTest.java]: Should this be userRep.getUsername()?

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/UserTest.java [hunk 2]
```diff
@@ -1458,6 +1470,25 @@ public void updateUserWithoutUsername() {
...
 
+    @Test
+    public void updateUserWithEmailAsUsername() {
+        switchRegistrationEmailAsUsername(true);
+
+        String id = createUser();
+
+        UserResource user = realm.users().get(id);
+        UserRepresentation userRep = user.toRepresentation();
+        assertEquals("user1@localhost", userRep.getEmail());
```
Comment (stianst) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/UserTest.java]: Should this be userRep.getUsername()?

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/UserTest.java [hunk 3]
```diff
@@ -1458,6 +1470,25 @@ public void updateUserWithoutUsername() {
...
+
+        userRep.setEmail("user11@localhost");
+        updateUser(user, userRep);
+
+        userRep = realm.users().get(id).toRepresentation();
+        assertEquals("user11@localhost", userRep.getEmail());
```
Comment (stianst) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/UserTest.java]: Should this be userRep.getUsername()?

**Review by stianst — CHANGES_REQUESTED** [review]

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/UserTest.java (hunk 1 above)
Comment (douglaspalmer) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/UserTest.java]: Yes, fixed, thanks

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/UserTest.java (hunk 2 above)
Comment (douglaspalmer) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/UserTest.java]: Yes, fixed, thanks.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/UserTest.java (hunk 3 above)
Comment (douglaspalmer) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/UserTest.java]: Yes, fixed, thanks.

**Review by stianst — APPROVED** [review]

**Pull request merged**